crates/subspace-farmer/src/single_disk_plot/farming.rs

```diff
@@ -0,0 +1,176 @@
+use crate::single_disk_plot::{FarmingError, SectorMetadata};
+use bitvec::prelude::*;
+use parity_scale_codec::{Decode, IoReader};
+use schnorrkel::Keypair;
+use std::io;
+use std::io::SeekFrom;
+use subspace_core_primitives::crypto::blake2b_256_254_hash;
+use subspace_core_primitives::crypto::kzg::Witness;
+use subspace_core_primitives::{
+    Blake2b256Hash, Chunk, Piece, PublicKey, SectorId, Solution, SolutionRange, PIECE_SIZE,
+};
+use subspace_rpc_primitives::FarmerProtocolInfo;
+use subspace_solving::{create_chunk_signature, derive_chunk_otp};
+use subspace_verification::is_within_solution_range;
+use tracing::error;
+
+/// Sector that can be used to create a solution that is within desired solution range
+#[derive(Debug, Clone)]
+pub struct EligibleSector {
```

Comment:
Just `Sector` conveys the same meaning no?

Reply:
I chose `EligibleSector` because it is information about the sector that can be used in solution, not the sector itself (its contents). Just wanted to prevent potential confusion.